src/apps/dictionaries/components/DictionaryForm.tsx

```diff
@@ -83,10 +86,33 @@ const DictionaryForm: React.FC<Props> = ({
   usersOrgs,
   errors,
   context = CONTEXT.view,
-  savedValues
+  savedValues,
+  createAndAddConcepts
 }) => {
   const classes = useStyles();
 
+  const { state = { description: '', references: [], default_locale: '', supported_locales: [], owner_url: '', preferred_source: '', public_access: ''} } = useLocation() || {};
+  const { copying } = useQueryParams<{
+    copying: string;
+  }>();
+  const { 
+    description,
+    references, 
+    default_locale, 
+    supported_locales, 
+    owner_url, 
+    preferred_source, 
+    public_access 
+  } = state;
+
+  const [copy, setCopy] = useState({...initialValues, description, references, default_locale, supported_locales: supported_locales !== null ? supported_locales : [], owner_url, preferred_source, public_access});
+
+  useEffect(() => {
+    if (copying !== '') {
+      setCopy({...initialValues, description, references, default_locale, supported_locales: supported_locales !== null ? supported_locales : [], owner_url, preferred_source, public_access})
+    }
+  }, [copying]); //eslint-disable-line
+
```

Comment:
Thanks @ibacher 
1. About writing it like `const copy = {...initialState, ...state}` it replaces all the values in the `initialValues` object hence repopulating everything which we don't want, Also it sends the id while submit which throws an error of an existing value, that's why I picked what we want.
2. For adding a copying query param in the URL, it helps when we are determining which action to dispatch since while copying we add another action of `addingConcepts`, but I believe it can be made better following option (1).

Reply:
@hadijahkyampeire Re 1. That's a good point. Here's a slightly better proposal:

In `types.ts`:
```ts
import { pick } from "lodash"

// ...

export type CopyableDictionary = Pick<
  APIDictionary,
  'description'       |
  'default_locale'    |
  'supported_locales' |
  'owner_url'         |
  'preferred_source'  |
  'public_access'     |
  'references'
>

export const dictionaryToCopyableDictionary: CopyableDictionary = (
  dictionary: APIDictionary
) => pick(dictionary, [
  'description',
  'default_locale',
  'supported_locales',
  'owner_url',
  'preferred_source',
  'public_access',
  'references'
]);
```

Then, when loading the dictionary, use something like:

```ts
const copy = { ...initialValues, ...dictionaryToCopyableDictionary(dictionary) } as APIDictionary;
```

That can be modified for whatever subset of properties we want to allow (I would suggest that even `owner_url` and `public_access` are probably things we shouldn't copy, but I'll defer to how people want this feature to work.).